actionpack/test/dispatch/request/jsonapi_params_parsing_test.rb

```diff
@@ -0,0 +1,53 @@
+require 'abstract_unit'
+
+class JsonapiParamsParsingTest < ActionDispatch::IntegrationTest
+  class TestController < ActionController::Base
+    class << self
+      attr_accessor :last_request_parameters
+    end
+
+    def parse
+      self.class.last_request_parameters = request.request_parameters
+      head :ok
+    end
+  end
+
+  def teardown
+    TestController.last_request_parameters = nil
+  end
+
+  test "parses json params for application json" do
+    payload = "{ \"data\": { \"type\": \"posts\", \"attributes\": { \"title\":"\
+              "\"Hello\", \"date\": \"today\"} , \"relationships\": { "\
+              "\"author\": { \"data\": { \"id\": \"2\", \"type\": \"users\" }"\
+              " }, \"comments\": { \"data\": [ { \"type\": \"comments\","\
+              "\"id\": \"3\" }, { \"type\": \"comments\", \"id\": \"4\" } ]"\
+              " }, \"journal\": { \"data\": null } } } }"
+    expected = { "_type" => "posts", "title" => "Hello", "date" => "today",
+                 "author_id" => "2", "author_type" => "User",
+                 "comment_ids" => ["3", "4"], "comment_types" => ["Comment", "Comment"],
```

Comment:
Interesting, since https://github.com/rails/rails/blob/1e791705dcea650a034b7ccb4d922f455648202d/activerecord/lib/active_record/serialization.rb#L11 is should be basically the same.  I wonder how used these APIs are